snapcraft/internal/build_providers/_lxd/_lxd.py

```diff
@@ -365,6 +382,49 @@ def pull_file(self, name: str, destination: str, delete: bool = False) -> None:
     def shell(self) -> None:
         self._run(command=["/bin/bash"])
 
+    def _wait_for_network(self) -> None:
+        while True:
+            if self._container is None:
+                raise RuntimeError("Attempted to use container before starting.")
+            if any(
+                a["address"]
+                for a in self._container.state().network["eth0"]["addresses"]
+                if a["family"] == "inet"
```

Comment:
Is ipv6 supported today?  or will lxd tunnel 4to6?

Reply:
there's always an ipv6 address assigned, so this is not the best mechanism to have if the network is ipv6 only and we might need to fallback to connectivity checks (but that would make all builds require being online). Not sure what is best at this point with this. So far, for snapcraft we have been recommending lxd defaults, which setup ipv4

Reply:
@stgraber  or @Saviq do you have suggestions for this?

Reply:
LXD sets up a private IPv4 and IPv6 subnet by default.
As we use a private IPv6 subnet, the C library will prefer IPv4 over it, however should connecting over IPv4 fail, IPv6 will be attempted.

Reply:
Should we wait for both? Or just get started once ipv4 is configured?

Reply:
The normal recommend is to wait for what you care about. In most cases you don't care about an address, you care about being able to connect to stuff.

In our own CI, our go-to test is to run `getent hosts linuxcontainers.org` in the container. When that succeeds it means we have an IP and DNS working, doesn't matter if over v4 or v6.